### PR TITLE
s4708525 - Classify Alzheimer’s disease (normal and AD) of the ADNI brain data using Convolutional Vision Transformer (CvT)

### PR DESCRIPTION
**Pull request by yhandoko**
This project aims to classify Alzheimer's disease (normal vs. AD) using the Convolutional Vision Transformer (CvT) on ADNI brain data.  This project concludes that the Convolutional Vision Transformer can effectively perform image classification on the ADNI brain data. However, improvements are needed in this architecture to enhance the model's generalization and robustness. The accuracy of the test data was 66.26%, which is lower than the 74.1% accuracy of the validation data, indicating slight overfitting in this experiment.

Name: Yuda Hendriawan Budi Handoko
Student ID: s4708525

File structures:

`results`: this folder contains the results of the running code, including the visualization of the loss during training, including the image resource to build the README.md file

`README.md`: this file contains the overview of the project and a description of the project results

`dataset.py`: this file handles the data loading, including the transformation and augmentation.

`modules.py`: this file defines the Convolutional Vision Transformer model.

`predict.py`: This file runs an example of how to use the generated model from `train.py` to predict ADNI dataset classes.

`train.py`: This file performs the training process using the Convolutional Vision Transformer model and saves the model.

`utils.py`: this code contains the function to do a visualization

### CODE REVIEW

**Comment by yhandoko** [comment]
Hi @prprakash02 , thank you for the feedback; I was setting the epochs for running to 100, but it was only showing 40 because of the early stopping mechanism. Maybe I should explain that, shouldn't I?

**Comment by prprakash02** [comment]
Hi,

Can you please add that in the main part of the README file (about the early stopping mechanism) since it was slightly unclear!

Thanks

**Comment by prprakash02** [comment]
**UPDATE:**

Recognition Problem: Total (18/20)

**Comment by yhandoko** [comment]
So, will I get full mark for the incorporating feedback part so far?

**Comment by prprakash02** [comment]
For pull request, there will be someone from the tutoring team marking again, if they have any feedback you might have to address that as well.

Finallyz after all formal marking you will be awarded if incorporated :)

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request should  include a clear description about the file structure .
No previous feedback incorporated .

File Organizing: Well-organized files.

Commit Log:
Commit messages are progressive for the Recognition Problem solved using 4 files .
Commits are regularly made.

Documentation:
Code comments and docstrings are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about future improvements in the ReadMe.

**Comment by yhandoko** [comment]
Hi @aniketgupta17, Thank you for your feedback. 

I have updated the pull request description and added the future improvements part in the README file.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Some inconsistancies of code and report noted.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard. ViT (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-3</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by yhandoko** [comment]
Hi @gayanku ,  thank you for giving me the feedback.
May I know what is the meaning of 'No Header Block' ?

Thank you.
